File: spa/src/app/model/project-model.ts
export interface ProjectModel {
  id?: number;
  name: string;
  createdDate: Date;
  deadLineDate?: Date;
  lastUpdatedDate: Date;
  finishedDate?: Date;
  sellPrice?: number;
  capital: number;
  fail: boolean;
  finish: boolean;
  profitInPersen: number;
  description: string;
}

export const initProjectModel = () => {
  return {
    name: '',
    createdDate: new Date(),
    deadLineDate: new Date(),
    lastUpdatedDate: new Date(),
    finishedDate: new Date(),
    sellPrice: 0.0,
    capital: 0.0,
    fail: false,
    finish: false,
    profitInPersen: 10,
    description: ''
  } as ProjectModel
}

File: api/Model/ProjectModel.cs
using TypeGen.Core.TypeAnnotations;

namespace Planerp.Model;

[ExportTsInterface(OutputDir = Constant.TypeGenBaseDirectory)]
public class Project : BaseModel
{
    public string Name { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime? DeadLineDate { get; set; }
    public DateTime LastUpdatedDate { get; set; }
    public DateTime? FinishedDate { get; set; }
    public float? SellPrice { get; set; }
    public float Capital { get; set; }
    public bool Fail { get; set; }
    public bool Finish { get; set; }
    public float? ProfitInPersen { get; set; }
    public string? Description { get; set; }

    // public List<string> ListComponent { get; set; }

    public int? LoggerModelId { get; set; }
    public int? ComponentId { get; set; }
}
